### PR TITLE
Issue #12379: Checker false violation removals on other failures

### LINKED ISSUE #12379: Checker: false violation removals on other failures
Identified at https://github.com/checkstyle/checkstyle/actions/runs/3405402498/jobs/5663345165 ,

`xml-maven-plugin` failed and checker reported back all suppressions should be removed even though Checker never actually run.

We should skip anything else besides Checker to ensure groovy script behaves correctly.

### PR DESCRIPTION
**Pull request by romani**
fixes #12379

### CODE REVIEW

File: .ci/checker-framework.groovy [hunk 1]
```diff
@@ -112,7 +112,8 @@ private static int checkCheckerFrameworkReport(final String profile) {
  */
 private static List<List<String>> getCheckerFrameworkErrors(final String profile) {
     final List<String> checkerFrameworkLines = new ArrayList<>()
-    final String command = "mvn -e --no-transfer-progress clean compile -P${profile}"
+    final String command = "mvn -e --no-transfer-progress clean compile"
+        + " -P${profile},no-validations"
```
Comment (rnveach) [inline .ci/checker-framework.groovy:116]: CI is failing, `+` has to be on previous line.
https://github.com/checkstyle/checkstyle/actions/runs/4097261782/jobs/7065502375#step:6:15

**Review by rnveach — CHANGES_REQUESTED** [review]

File: .ci/checker-framework.groovy (hunk 1 above)
Comment (romani) [reply .ci/checker-framework.groovy:116]: Yes, we should disable this rule one day in future.

Comment (rnveach) [reply .ci/checker-framework.groovy:116]: My understanding was that this is something in groovy, not a check we turned on.
https://stackoverflow.com/a/5079914/1016482

Comment (rnveach) [reply .ci/checker-framework.groovy]: I rebased and fixed the code.

**Comment by rnveach** [comment]
CI is all green.

https://github.com/checkstyle/checkstyle/actions/runs/4116348792/jobs/7106284737#step:6:64
Checker still ran and looks like it is outputting the expected information.

**Review by rnveach — APPROVED** [review]

File: .ci/checker-framework.groovy (hunk 1 above)
Comment (romani) [reply .ci/checker-framework.groovy]: thanks a lot, I just pushed a fix and got surprised that I have no PR

**Pull request merged**